### PR TITLE
feat(kbadge): introduce kui tokens [KHCP-7694]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-7694

Changes:
- introduce `kui-` design tokens in `KBadge` component
- break down usage of Kongponents utility classes in `KBadge` component template AND in documentation markdown file

## PR Checklist

* [ ] Does not introduce dependencies
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/badge.md [hunk 1]
```diff
@@ -22,12 +22,14 @@ The Badge component can take the following appearance values:
 - `neutral`
 - `custom`
 
-<KBadge appearance="success" class="mr-2">SUCCESS</KBadge>
-<KBadge appearance="warning" class="mr-2">WARNING</KBadge>
-<KBadge appearance="danger" class="mr-2">DANGER</KBadge>
-<KBadge appearance="info" class="mr-2">INFO</KBadge>
-<KBadge appearance="neutral" class="mr-2">NEUTRAL</KBadge>
-<KBadge>DEFAULT</KBadge>
+<div class="vertical-spacing">
```
Comment (adamdehaven) [inline docs/components/badge.md:25]: nit: this class name doesn't make sense -- shouldn't it be horizontal?

File: src/components/KBadge/KBadge.vue [hunk 2]
```diff
@@ -50,26 +50,11 @@ import { ref, computed, watch, useAttrs, PropType } from 'vue'
 import KButton from '@/components/KButton/KButton.vue'
 import KIcon from '@/components/KIcon/KIcon.vue'
 import KTooltip from '@/components/KTooltip/KTooltip.vue'
-import type { BadgeAppearance, BadgeAppearanceRecord, BadgeShapeRecord, BadgeShape } from '@/types'
+import { BadgeAppearance, BadgeShape, appearances, shapes } from '@/types'
```
Comment (adamdehaven) [inline src/components/KBadge/KBadge.vue:53]: why did you remove `import type` ?
```suggestion
import type { BadgeAppearance, BadgeShape, appearances, shapes } from '@/types'
```

File: src/components/KBadge/KBadge.vue [hunk 3]
```diff
@@ -50,26 +50,11 @@ import { ref, computed, watch, useAttrs, PropType } from 'vue'
...
-  info: 'info',
-  warning: 'warning',
-  custom: 'custom',
-  neutral: 'neutral',
-}
-
-export const shapes: BadgeShapeRecord = {
-  rounded: 'rounded',
-  rectangular: 'rectangular',
-}
```
Comment (adamdehaven) [inline src/components/KBadge/KBadge.vue:71]: question: These exports were moved to the `badge.ts` file; however, I don't see them being used in this file anymore...? 

Also, why are you making type interface changes on a PR for tokens? This should likely be reverted as the changes are unrelated.

File: src/types/badge.ts [hunk 4]
```diff
@@ -3,3 +3,18 @@ export type BadgeAppearanceRecord = Record<BadgeAppearance, BadgeAppearance>
...
+  info: 'info',
+  warning: 'warning',
+  custom: 'custom',
+  neutral: 'neutral',
+}
+
+export const shapes: BadgeShapeRecord = {
+  rounded: 'rounded',
+  rectangular: 'rectangular',
+}
```
Comment (adamdehaven) [inline src/types/badge.ts:20]: nit: no commas are needed

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: src/types/badge.ts (hunk 4 above)
Comment (portikM) [reply src/types/badge.ts]: Eslint shows error if removed

File: docs/components/badge.md (hunk 1 above)
Comment (portikM) [reply docs/components/badge.md]: Fixed

File: src/types/badge.ts [hunk 5]
```diff
@@ -3,3 +3,18 @@ export type BadgeAppearanceRecord = Record<BadgeAppearance, BadgeAppearance>
...
+  info: 'info',
+  warning: 'warning',
+  custom: 'custom',
+  neutral: 'neutral',
+}
+
+export const BadgeShapes: BadgeShapeRecord = {
+  rounded: 'rounded',
+  rectangular: 'rectangular',
+}
```
Comment (adamdehaven) [inline src/types/badge.ts:20]:
```suggestion
export const BadgeAppearances: BadgeAppearanceRecord = {
  default: 'default',
  success: 'success',
  danger: 'danger',
  info: 'info',
  warning: 'warning',
  custom: 'custom',
  neutral: 'neutral',
} as const

export const BadgeShapes: BadgeShapeRecord = {
  rounded: 'rounded',
  rectangular: 'rectangular',
} as const
```

**Review by adamdehaven — APPROVED** [review]
lgtm
